docs/source/get-started/building-from-source.rst

```diff
@@ -0,0 +1,235 @@
+        grep kokkos-${KOKKOS_VERSION}.tar.gz kokkos-${KOKKOS_VERSION}-SHA-256.txt | shasum -c
+
+
+    The output should be ``kokkos-4.5.01.tar.gz: OK`` (or similar, depending on
+    the version).  If the checksum doesn't match, **do not use the downloaded
+    archive** as it may be corrupted or tampered with.
+
+4.  **Extract the Archive:**  Once you've verified the checksum, extract the archive:
+
+    .. code-block:: sh
+
+        tar -xzvf kokkos-${KOKKOS_VERSION}.tar.gz
+
+Cloning the Git Repository (For Development Versions)
+-----------------------------------------------------
+
+If you need the latest features or want to contribute to Kokkos, you can clone
+the Git repository.
+
+1.  **Clone the Repository:**
+
+    .. code-block:: sh
+
+        git clone https://github.com/kokkos/kokkos.git
+
+    This will clone the repository into a directory named ``kokkos``.
+
+2.  **Check Out a Release Tag (Recommended for Development):**
+    While the ``develop`` branch is generally kept stable, it's still under
+    active development.  For more predictable behavior, check out a specific
+    release tag:
+
+    .. code-block:: sh
+
+        cd kokkos
+        git checkout 4.5.01  # Replace with the desired version tag
+
+    To see available tags:
+
+    .. code-block:: sh
+
+        git tag
+
+    Or, to stay on the bleeding edge (use with caution):
+
+    .. code-block:: sh
+
+        git checkout develop
+
+
+Which Method Should I Use?
+--------------------------
+
+* **Tagged Releases:** Use this method unless you have a specific reason to use
+  a development version.  Tagged releases are the most stable and well-tested.
+* **Git Repository (Development Versions):** Use this method if you need the
+  very latest features, want to contribute to Kokkos, or need to debug a
+  specific issue that's been fixed in the development branch.  Be aware that
+  development versions may be less stable than releases.
+
+No matter which method you choose, always verify the integrity of the
+downloaded source code.  This is a crucial security practice.
+
+
+Configuring and Building Kokkos
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This section describes how to configure and build Kokkos.  We assume you are in
+the root directory of the Kokkos source code (or the project embedding Kokkos).
+
+Configuring Kokkos
+------------------
+
+Use the following command to configure Kokkos:
+
+.. code-block:: sh
+
+    cmake -B builddir [<options...>]
+
+
+``-B builddir`` creates a ``builddir`` directory named build (you can choose a
+different name if you prefer).  Kokkos requires out-of-source builds.  The
+``[<options...>]`` part is where you specify the configuration options.
+
+**Common CMake Options**
+
+These options are generally useful for any CMake project:
+
+* ``-DCMAKE_CXX_COMPILER=<compiler>``: Specifies the full path to the C++
+  compiler. For example, use ``hipcc`` for AMD GPUs, ``icpx`` for Intel GPUs,
+  or ``g++`` or ``clang++`` for CPUs.
```

Comment:
We may wanna say something about CUDA here. There are so many examples, but NVIDIA is missing. A user could assume that simply means he should set NVCC as the CMAKE_CXX_COMPILER since one uses the vendor compilers for the AMD and Intel GPUs.

Reply:
Let's improve in a follow up